### DISCUSSION TITLE
Add automatic "drill-down" views

### DISCUSSION THREAD

**Discussion by pavelpykhtin** [likec4/likec4#1054] [Ideas]
Hi, @davydkov ,

There is a common task - examine relationships of one particular component X. In most cases it is enough to have a scoped view (view of X { include *; }). And it seems that this view can be build "on the fly" - no need to define it manually. Thus, each component will have a navigation button: if there is no explicitly defined view for this component, then an automatically generated view will be shown.

Those automatic views should not allow manual editing and provide link to source file.

There is a downside, of course: links to "core" views (those which provide most information) are lost among the links to automatic views. Thus, the toggle in configuration required.

**Comment by davydkov** [comment]
Hi @pavelpykhtin 

Something like "dependencies view" in icepanel?

![SCR-20241008-ljzp](https://github.com/user-attachments/assets/62d8fd67-99d3-4f5d-8e46-c50ffbc30a44)

**Reply by pavelpykhtin** [reply]
Yes, exactly

**Reply by davydkov** [reply]
Experiments in v1.13 - [template.likec4.dev](https://template.likec4.dev/view/cloud) or [Big Bank](https://likec4.dev/examples/bigbank/)

![SCR-20241018-didx](https://github.com/user-attachments/assets/6565e3da-3f9e-4991-8d2c-e28b88c94375)

What do you think?

**Reply by aelberg** [reply]
Trying this out with my existing model this morning.  Amazing!!!  Thank You!

**Reply by pavelpykhtin** [reply]
@davydkov ,  It's awesome! Thank you!

**Reply by davydkov** [reply]
@pavelpykhtin @aelberg 
I played with relationship decomposition (like those with `[...]`)

![SCR-20241020-dgyj](https://github.com/user-attachments/assets/3983d564-bc7a-4f98-9509-5788064ce7cb)

**Reply by pavelpykhtin** [reply]
Seems pretty good! Will it replace the existing relation popup or will be opened with the command from that popup?

**Reply by aelberg** [reply]
Loving the relationship decomposition as well!  Thoughts on including the icons in these views as well?

**Comment by aelberg** [comment]
@davydkov is there a way to specify a viewId to directly include these views with <LikeC4View viewId="" />?
